api/sip/__init__.py

```diff
@@ -47,17 +49,30 @@ def __init__(self, server, port, login_user_id,
 
         """
         super(SIP2AuthenticationProvider, self).__init__(**kwargs)
-        if client:
-            self.client = client
-        else:
-            self.client = SIPClient(
-                target_server=server, target_port=port,
-                login_user_id=login_user_id, login_password=login_password,
-                location_code=location_code, separator=field_separator
+        try:
+            if client:
+                if callable(client):
+                    client = client()
```

Comment:
If it's not callable then it's the actual client object, not a client class that needs instantiation.

In most tests I pass in the actual mock client, but I need this feature to test the client that blows up when you call connect(). Since that's called during the constructor, I had to make sure that the constructor was called inside the try/catch block.